### PR TITLE
kv, puller (ticdc): remove old kvClient and puller

### LINKED ISSUE #10440: [CDC] Remove old Puller code base
Since we already by default enable multiplexingPuller in `SourceManager`, the old `tablePullers` is not used any more.

### PR DESCRIPTION
**Pull request by asddongmen**
### What problem does this PR solve?

Issue Number: close #10440 

#### Tests 

 - No code

### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/puller/multiplexing_puller.go [hunk 1]
```diff
@@ -71,6 +73,14 @@ type tableProgressWithSubID struct {
 	subID kv.SubscriptionID
 }
 
+// Puller pull data from tikv and push changes into a buffer.
```
Comment (hicqu) [inline cdc/puller/multiplexing_puller.go:76]: Seems it can be removed.

**Review by hicqu — APPROVED** [review]

**Review by CharlesCheung96 — APPROVED** [review]

File: cdc/puller/multiplexing_puller.go (hunk 1 above)
Comment (asddongmen) [reply cdc/puller/multiplexing_puller.go:76]: Thank you, I will remove it in next code improvement pull request.

**Pull request merged**